### PR TITLE
For expression quote and scope

### PR DESCRIPTION
**Pull request by bloderxd**
This PR introduces `KtForExpression` Quote and its scope mapping. This PR is related to https://github.com/arrow-kt/arrow-meta/issues/89

### Checklist for `KtForExpression`
 - [x] Added/replaced the inverse element in ElementScope
 - [x] Destructure and make available all methods related to rendering properties, but no logic
 - [x] Add documentation for element

### CODE REVIEW

File: compiler-plugin/src/main/kotlin/arrow/meta/quotes/ForExpression.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  quote(match, map) { ForExpressionScope(it) }
+
+/**
+ * A template destructuring [Scope] for a [KtForExpression]
+ */
+class ForExpressionScope(
+  override val value: KtForExpression,
+  val `(parameter)`: ParameterScope = ParameterScope(value.loopParameter),
+  val range: Scope<KtExpression> = Scope(value.loopRange),
+  val destructuringDeclaration: Scope<KtDestructuringDeclaration> = Scope(value.destructuringDeclaration)
```
Comment (bloderxd) [inline compiler-plugin/src/main/kotlin/arrow/meta/quotes/ForExpression.kt]: this should be replaced by `KtDestructuringDeclaration` proper scope when it's implemented

Comment (ahinchman1) [reply compiler-plugin/src/main/kotlin/arrow/meta/quotes/ForExpression.kt]: Thanks! Added it to #89

File: compiler-plugin/src/main/kotlin/arrow/meta/quotes/ForExpression.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * @param match designed to to feed in any kind of [KtForExpression] predicate returning a [Boolean]
+ * @param map map a function that maps over the resulting action from matching on the transformation at the PSI level.
+ *
+ * forExpression({ true }) { e ->
+ *   Transform.replace(
+ *    replacing = e,
+ *    newDeclaration = """ for $`(param)` { $body } """.`for`
+ *   )
+ *  }
```
Comment (ahinchman1) [inline compiler-plugin/src/main/kotlin/arrow/meta/quotes/ForExpression.kt]: Here' s a cool thing - we can add ``` here like:

```
 *``````kotlin:ank:silent
 * forExpression({ true }) { e ->
 *   Transform.replace(
 *    replacing = e,
 *    newDeclaration = """ for $`(param)` { $body } """.`for`
 *   )
 *  }
 * ```
```

Comment (ahinchman1) [reply compiler-plugin/src/main/kotlin/arrow/meta/quotes/ForExpression.kt]: Awesome job!

File: compiler-plugin/src/main/kotlin/arrow/meta/quotes/Parameter.kt [hunk 3]
```diff
@@ -20,12 +20,12 @@ fun Meta.parameter(
...
-  val `(typeParameters)`: ScopedList<KtTypeParameter> = ScopedList(prefix = "<", value = value.typeParameters, postfix = ">"),
-  val defaultValue: KtExpression? = value.defaultValue
+  override val value: KtParameter?,
+  val modality: Name? = value?.modalityModifierType()?.value?.let(Name::identifier),
+  val visibility: Name? = value?.visibilityModifierType()?.value?.let(Name::identifier),
+  val name: Name? = value?.nameAsName,
+  val `@annotations`: ScopedList<KtAnnotationEntry> = ScopedList(value?.annotationEntries ?: listOf()),
+  val type: Scope<KtTypeReference> = Scope(value?.typeReference),
+  val `(typeParams)`: ScopedList<KtTypeParameter> = ScopedList(prefix = "<", value = value?.typeParameters ?: listOf(), postfix = ">"),
+  val defaultValue: Scope<KtExpression> = Scope(value?.defaultValue)
```
Comment (ahinchman1) [inline compiler-plugin/src/main/kotlin/arrow/meta/quotes/Parameter.kt]: Thanks for going back on this! Maybe we can do a separate PR for whatever naming convention cleanups there are after this one
